frontend/src/app/shared/components/suggested-challenge/suggested-challenge.component.ts

```diff
@@ -1,47 +1,103 @@
-import { Component, Input } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { BaseComponent } from '@core/base/base.component';
+import { ChallengeService } from '@core/services/challenge.service';
+import { LanguageService } from '@core/services/language.service';
+import { ToastrNotificationsService } from '@core/services/toastr-notifications.service';
+import { ChallengePreview } from '@shared/models/challenge/challenge-preview';
+import { SuggestionSettings } from '@shared/models/challenge/suggestion-settings';
+import { Language } from '@shared/models/language/language';
+import { takeUntil } from 'rxjs';
 
 @Component({
     selector: 'app-suggested-challenge',
     templateUrl: './suggested-challenge.component.html',
     styleUrls: ['./suggested-challenge.component.sass'],
 })
-export class SuggestedChallengeComponent {
-    @Input() tags: string[] = ['Algorithms', 'Strings', 'Data Types', 'Formatting', 'Logic', 'Graph', 'Stack'];
-
-    @Input() constraints: string[] = [
-        'Strings passed in will consist of only letters and spaces.',
-        'Spaces will be included only when more than one word is present.',
-        'Some another constraint.',
-        'Some another constraint.',
-        'Some another constraint.',
-    ];
+export class SuggestedChallengeComponent extends BaseComponent implements OnInit {
+    challenge?: ChallengePreview;
 
-    @Input() languages: string[] = [
-        'C#',
-        'Javascript',
-    ];
+    languagesNames: string[] = [];
+
+    languages: Language[] = [];
+
+    suggestionTypesNames: string[];
 
-    score: string[] = [
-        'Fundamentals',
-        'Rank up',
-        'Practice and Repeat',
-        'Beta',
-        'Random',
+    suggestionTypes = [
```

Comment:
why do we have hard coded Ids here?

Reply:
It corresponds to **enum** **SuggestionType** on backend
{
    Fundamentals,
    RankUp,
    PracticeAndRepeat,
    Beta,
    Random
}